crypto/err/err.c

```diff
@@ -222,26 +222,30 @@ static void build_SYS_str_reasons(void)
         ERR_STRING_DATA *str = &SYS_str_reasons[i - 1];
 
         str->error = ERR_PACK(ERR_LIB_SYS, 0, i);
-        if (str->string == NULL) {
+        /*
+         * If we have used up all the space in strerror_pool,
+         * there's no point in calling openssl_strerror_r()
+         */
+        if (str->string == NULL && cnt < sizeof(strerror_pool)) {
             if (openssl_strerror_r(i, cur, sizeof(strerror_pool) - cnt)) {
-                size_t l = strlen(cur);
+                size_t l = strlen(cur) + 1;
```

Comment:
I think Richard wanted this line to be reverted.

Reply:
And maybe the loop below can stay in the original form, just changing
`while (ossl_isspace(cur[-1])) {` to `while (cur > strerror_pool && ossl_isspace(cur[-1])) {'

Reply:
Yup

Reply:
Yes, this makes it work again.